### PR TITLE
[GSSDLC-126] Auto create WIP entries when a new project is created

### PR DESCRIPTION
**Pull request by jhonypg**

### CODE REVIEW

File: src/main/java/objective/taskboard/controller/ProjectController.java [hunk 1]
```diff
@@ -166,24 +176,42 @@
...
         if (!hasTeamsOnData)
             return;
         data.teams.forEach(team -> {
-            createTeamAndConfigurations(data.projectKey, team.name, data.teamLeader, data.teamLeader, team.members);
+            createTeamAndConfigurations(data.projectKey, team.name, data.teamLeader, data.teamLeader, team.members, team.wipConfigurations);
         });
     }
 
-    private void validateTeamsDontExist(ArrayList<ProjectCreationDataTeam> pcdTeams) {
+    private void validateTeamsAndWipConfigurations(ArrayList<ProjectCreationDataTeam> pcdTeams) {
```
Comment (herbert-venancio) [inline src/main/java/objective/taskboard/controller/ProjectController.java]: Dê preferencia a interface List no lugar da implementação

File: src/main/java/objective/taskboard/controller/ProjectController.java [hunk 2]
```diff
@@ -193,10 +221,12 @@ private void createProjectFilterConfiguration(String projectKey) {
         projectRepository.save(configuration);
     }
 
-    private void createTeamAndConfigurations(String projectKey, String teamName, String manager, String coach, ArrayList<String> members) {
+    private void createTeamAndConfigurations(String projectKey, String teamName, String manager, String coach, ArrayList<String> members, ArrayList<ProjectCreationDataWip> wipConfigurations) {
```
Comment (herbert-venancio) [inline src/main/java/objective/taskboard/controller/ProjectController.java]: Dê preferencia a interface List no lugar da implementação

File: src/main/java/objective/taskboard/controller/ProjectController.java [hunk 3]
```diff
@@ -217,6 +247,14 @@ private ProjectTeam createProjectTeam(String projectKey, TeamFilterConfiguration
         return projectTeamRepo.save(projectTeam);
     }
 
+    private void createTeamWipConfigurations(Team team, ArrayList<ProjectCreationDataWip> wipConfigurations) {
```
Comment (herbert-venancio) [inline src/main/java/objective/taskboard/controller/ProjectController.java]: Dê preferencia a interface List no lugar da implementação

File: src/main/java/objective/taskboard/controller/ProjectCreationData.java [hunk 4]
```diff
@@ -27,8 +27,15 @@
     public String projectKey;
     public String teamLeader;
     public ArrayList<ProjectCreationDataTeam> teams;
+
     public static class ProjectCreationDataTeam {
         public String name;
         public ArrayList<String> members;
+        public ArrayList<ProjectCreationDataWip> wipConfigurations;
```
Comment (herbert-venancio) [inline src/main/java/objective/taskboard/controller/ProjectCreationData.java]: Dê preferencia a interface List no lugar da implementação

File: src/main/java/objective/taskboard/controller/ProjectController.java [hunk 5]
```diff
@@ -166,24 +176,42 @@
...
+        if (teamRepository.exists(teamName))
+            throw new IllegalArgumentException("Team '" + teamName + "' already exists.");
+    }
+
+    private void validateWipConfigurationsDontExist(ProjectCreationDataTeam team) {
+        List<WipConfiguration> wipConfigurations = wipConfigRepo.findByTeam(team.name);
+        for (ProjectCreationDataWip newWipConfig : team.wipConfigurations) {
+            String statusName = metadataService.getStatusById(newWipConfig.statusId).getName();
+            for (WipConfiguration wipConfiguration : wipConfigurations) {
+                if (wipConfiguration.getStatus() == statusName);
```
Comment (objective-solutions-user) [inline src/main/java/objective/taskboard/controller/ProjectController.java]: MAJOR Use the "equals" method if value comparison was intended. [rule](http://10.44.1.250:9000/coding_rules#rule_key=squid%3AS1698)
MINOR Remove this empty statement. [rule](http://10.44.1.250:9000/coding_rules#rule_key=squid%3AEmptyStatementUsageCheck)

**Comment by objective-solutions-user** [comment]
SonarQube analysis reported 2 issues
* ![MAJOR][MAJOR] 1 major
* ![MINOR][MINOR] 1 minor

Watch the comments in this conversation to review them.

[MAJOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR'
[MINOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR'

File: src/main/java/objective/taskboard/controller/ProjectController.java [hunk 6]
```diff
@@ -166,24 +176,42 @@
...
+        if (teamRepository.exists(teamName))
+            throw new IllegalArgumentException("Team '" + teamName + "' already exists.");
+    }
+
+    private void validateWipConfigurationsDontExist(ProjectCreationDataTeam team) {
+        List<WipConfiguration> wipConfigurations = wipConfigRepo.findByTeam(team.name);
+        for (ProjectCreationDataWip newWipConfig : team.wipConfigurations) {
+            String statusName = metadataService.getStatusById(newWipConfig.statusId).getName();
+            for (WipConfiguration wipConfiguration : wipConfigurations) {
+                if (wipConfiguration.getStatus().equals(statusName));
```
Comment (objective-solutions-user) [inline src/main/java/objective/taskboard/controller/ProjectController.java]: MINOR Remove this empty statement. [rule](http://10.44.1.250:9000/coding_rules#rule_key=squid%3AEmptyStatementUsageCheck)

**Comment by objective-solutions-user** [comment]
SonarQube analysis reported 1 issue
* ![MINOR][MINOR] 1 minor

Watch the comments in this conversation to review them.

[MINOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR'

**Pull request merged**